src/librustc/diagnostics.rs

```diff
@@ -335,6 +392,275 @@ This error indicates that an attempt was made to divide by zero (or take the
 remainder of a zero divisor) in a static or constant expression.
 "##,
 
+E0022: r##"
+Constant functions are not allowed to mutate anything. Thus, binding to an
+argument with a mutable pattern is not allowed. For example,
+
+```
+const fn foo(mut x: u8) {
+    // do stuff
+}
+```
+
+is bad because the function body may not mutate `x`.
+
+Remove any mutable bindings from the argument list to fix this error. In case
+you need to mutate the argument, try lazily initializing a global variable
+instead of using a const fn, or refactoring the code to a functional style to
+avoid mutation if possible.
+"##,
+
+
+E0038: r####"
+
+Trait objects like `Box<Trait>` can only be constructed when certain
+requirements are obeyed by the trait in question.
```

Comment:
This is seems like an awkward wording. I feel like you satisfy requirements or obey restrictions. But obey requirements feels weird.